src/libraries/System.Net.HttpListener/src/System/Net/Windows/HttpListener.Windows.cs

```diff
@@ -451,13 +451,16 @@ private unsafe void CloseRequestQueueHandle()
                 if (NetEventSource.IsEnabled) NetEventSource.Info($"Dispose ThreadPoolBoundHandle: {_requestQueueBoundHandle}");
                 _requestQueueBoundHandle?.Dispose();
                 _requestQueueHandle.Dispose();
+                // CancelIoEx is called after Dispose to prevent a race condition when GetContext and consequently
```

Comment:
Suggestions:

Add blank line before this comment block.

Better wording to make the grammar more understandable:

```c#
// CancelIoEx is called after Dispose to prevent a race condition involving parallel GetContext and //HttpReceiveHttpRequest calls.  Otherwise, calling CancelIoEx before Dispose might block the synchronous
// GetContext call until the next request arrives.
```